src/plugins/dam/components/asset-screen/DetailsTab.jsx

```diff
@@ -0,0 +1,53 @@
+import React, { lazy, Suspense } from 'react';
+import startCase from 'lodash-es/startCase';
+import { Card, CardBody, CardHeader } from 'reactstrap';
+import {
+  TextField
+} from 'components';
+import filesize from 'filesize';
+import TaggableFields from 'plugins/common/components/taggable-fields';
+
+const getFileSize = (value) => {
+    return filesize(value.toString(), {round :1});
+  }
+
+
+
+
+const components = {};
+const resolveComponent = (label) => {
+  if (components[label]) {
+    return components[label];
+  }
+
+  const file = startCase(label).replace(/\s/g, '');
+  components[label] = lazy(() => import(`./${file}Fields`));
+  return components[label];
+};
+
+export default function DetailsTab(props) {
+  const { label, node, nodeRef } = props;
+  const FieldsComponent = resolveComponent(label);
+  const schema = node.schema();
+
+  const getDimensions = (value) =>{
+    return `${value} x ${node.get('height')}`;
+ }
+
+  return (
+    <>
+      <Card>
+        <CardHeader>{startCase(label).replace(/\s/g, ' ')}</CardHeader>
+        <CardBody>
+          <TextField name="mime_type" label="MIME type" readOnly />
+          <TextField name="file_size" label="File size"  format={getFileSize} />
+          {label==='image-asset' && (
+            <TextField name="width" label="Dimensions"  format={getDimensions} />
```

Comment:
@meghedi considering assets have different fields we probably want a solution like teasers where the details if any can be generic but then include asset specific fields in its own component.  prefer to not have mixed fields here and there with if blocks around them.

Reply:
@gdbrown created a CommonFields.jsx file which holds the common ones.